src/main/java/org/folio/oaipmh/processors/MarcWithHoldingsRequestHelper.java

```diff
@@ -694,74 +692,82 @@ private HttpClientRequest createEnrichInventoryClientRequest(HttpClient httpClie
+  private void doPostRequestToSrs(SourceStorageSourceRecordsClient srsClient, Vertx vertx, boolean deletedRecordSupport,
+      List<String> listOfIds, AtomicInteger attemptsCount, int retryAttempts, Promise<Map<String, JsonObject>> promise) {
     try {
       srsClient.postSourceStorageSourceRecords("INSTANCE", deletedRecordSupport, listOfIds, srsResponse -> {
         int statusCode = srsResponse.statusCode();
         String statusMessage = srsResponse.statusMessage();
+        srsResponse.exceptionHandler(e -> {logger.error("SRS response error: " + e.getMessage(), e);
+            retrySRSRequest(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, retryAttempts, promise, statusCode, statusMessage);
+        });
+        if (statusCode >= 400) {
+          retrySRSRequest(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, retryAttempts, promise, statusCode, statusMessage);
+          return;
+        }
         if (statusCode != 200) {
-          if(statusCode >= 500) {
-            String warnMessage = "Got error response form SRS, status code: " + statusCode + ", status message: " + statusMessage;
-            logger.warn(warnMessage);
-            if (attemptsCount.get() <= 0) {
-              String errorMessage = "SRS didn't respond with expected status code after " + REQUEST_ATTEMPTS + " attempts. Canceling further request processing.";
-              srsClient.close();
-              handleException(promise, new IllegalStateException(errorMessage));
-              return;
-            }
-            logger.warn("Trying to request SRS again, attempts left: {}", attemptsCount.decrementAndGet());
-            vertx.setTimer(REREQUEST_SRS_DELAY, timer -> doPostRequestToSrs(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, promise));
-            return;
-          } else {
-            String errorMsg = getErrorFromStorageMessage("source-record-storage", srsResponse.request()
-              .absoluteURI(), srsResponse.statusMessage());
-            srsClient.close();
-            handleException(promise, new IllegalStateException(errorMsg));
-            return;
-          }
+          String errorMsg = getErrorFromStorageMessage("source-record-storage", srsResponse.request()
+            .absoluteURI(), srsResponse.statusMessage());
+          srsClient.close();
+          handleException(promise, new IllegalStateException(errorMsg));
+          return;
         }
-        srsResponse.bodyHandler(getSrsResponseBodyHandler(srsClient, promise));
-      });
+        srsResponse.bodyHandler(buffer -> handleSrsResponse(srsClient, promise, buffer));
+      }, e->  retrySRSRequest(srsClient, vertx, deletedRecordSupport, listOfIds, attemptsCount, retryAttempts, promise, 400, "Error in SRS response"));
```

Comment:
If I read this right, the exception handler provided here will only be invoked upon request exceptions (`request.exceptionHandler(...)`).  In this case there is no status code.  Furthermore, the message provided `Error in SRS response` is inaccurate.  `Error making SRS request` would be more accurate.